Tests/GeoLib/TestPolygon.cpp

```diff
@@ -138,40 +139,45 @@ TEST_F(PolygonTest, isPntInPolygonCheckOuterPoints)
 TEST_F(PolygonTest, containsSegment)
 {
 	{ // test segment (2,6)
-		GeoLib::Point const& a(*(_polygon->getPoint(2)));
-		GeoLib::Point const& b(*(_polygon->getPoint(6)));
-		ASSERT_FALSE(_polygon->containsSegment(a, b));
+		GeoLib::LineSegment const segment{
+		    const_cast<GeoLib::Point*>(_polygon->getPoint(2)),
+		    const_cast<GeoLib::Point*>(_polygon->getPoint(6))};
+		ASSERT_FALSE(_polygon->containsSegment(segment));
 	}
 
 	// test all segments of polygon
-	for (std::size_t k(0); k<_polygon->getNumberOfPoints()-1;k++) {
-		GeoLib::Point const& a = *(_polygon->getPoint(k));
-		GeoLib::Point const& b = *(_polygon->getPoint(k+1));
-		EXPECT_TRUE(_polygon->containsSegment(a, b));
+	for (auto segment_it(_polygon->begin()); segment_it != _polygon->end();
+	     ++segment_it)
+	{
+		EXPECT_TRUE(_polygon->containsSegment(*segment_it));
 	}
 
 	{ // 70
-		GeoLib::Point const& a = *(_polygon->getPoint(7));
-		GeoLib::Point const& b = *(_polygon->getPoint(0));
-		ASSERT_TRUE(_polygon->containsSegment(a, b));
+		GeoLib::LineSegment const segment{
+		    const_cast<GeoLib::Point*>(_polygon->getPoint(7)),
```

Comment:
The casts are not so nice. But probably they are a consequence of the design. Let's talk on monday.